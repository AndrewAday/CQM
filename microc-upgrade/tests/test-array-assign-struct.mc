struct foo {
  int i;
  float f;
}


int main()
{
  struct foo foo;
  struct foo foo2;
  struct foo[] foo_arr;
  foo_arr = make(struct foo, 1);
  foo = make(struct foo);
  foo.i = 1;
  foo.f = 3.14;
  print(foo.i);
  print_float(foo.f);

  foo_arr[0] = foo;
  print(foo.i);
  print_float(foo.f);
  foo.i = 2;
  foo.f = 4.12;
  foo2 = foo_arr[0];
  print(foo2.i);
  print_float(foo2.f);




  return 0;
}
