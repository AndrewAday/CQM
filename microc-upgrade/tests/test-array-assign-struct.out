1
3.140000
1
3.140000
2
4.120000
